cmd/generate.go

```diff
@@ -0,0 +1,294 @@
+		if name == "" {
+			return errors.New("--name is mandatory")
+		}
+		licenses, err := parseFlag(kind, license)
+		if err != nil {
+			return err
+		}
+		log.Debugf("parsed licenses: %+v", licenses)
+
+		images, err := parseFlag(kind, image)
+		if err != nil {
+			return err
+		}
+		log.Debugf("parsed images: %+v", images)
+
+		nodeDefs, err := parseNodesFlag(kind, nodes...)
+		if err != nil {
+			return err
+		}
+		log.Debugf("parsed nodes definitions: %+v", nodeDefs)
+
+		b, err := generateTopologyConfig(name, mgmtNetName, mgmtIPv4Subnet.String(), mgmtIPv6Subnet.String(), images, licenses, nodeDefs...)
+		if err != nil {
+			return err
+		}
+		log.Debugf("generated topo: %s", string(b))
+		if file != "" {
+			err = saveTopoFile(file, b)
+			if err != nil {
+				return err
+			}
+		}
+		if deploy {
+			reconfigure = true
+			if file == "" {
+				file = fmt.Sprintf("%s.yaml", name)
+				err = saveTopoFile(file, b)
+				if err != nil {
+					return err
+				}
+			}
+			topo = file
+			return deployCmd.RunE(deployCmd, nil)
+		}
+		if file == "" {
+			fmt.Println(string(b))
+		}
+		return nil
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(generateCmd)
+	generateCmd.Flags().StringVarP(&mgmtNetName, "network", "", "clab", "management network name")
+	generateCmd.Flags().IPNetVarP(&mgmtIPv4Subnet, "ipv4-subnet", "4", net.IPNet{}, "management network IPv4 subnet range")
+	generateCmd.Flags().IPNetVarP(&mgmtIPv6Subnet, "ipv6-subnet", "6", net.IPNet{}, "management network IPv6 subnet range")
+	generateCmd.Flags().StringSliceVarP(&image, "image", "", []string{}, "container image name, can be prefixed with the node kind. <kind>=<image_name>")
+	generateCmd.Flags().StringVarP(&kind, "kind", "", "srl", fmt.Sprintf("container kind, one of %v", supportedKinds))
+	generateCmd.Flags().StringSliceVarP(&nodes, "nodes", "", []string{}, "comma separated nodes definitions in format <num_nodes>:<kind>:<type>, each defining a Clos network stage")
+	generateCmd.Flags().StringSliceVarP(&license, "license", "", []string{}, "path to license file, can be prefix with the node kind. <kind>=/path/to/file")
+	generateCmd.Flags().StringVarP(&nodePrefix, "node-prefix", "", defaultNodePrefix, "prefix used in node names")
+	generateCmd.Flags().StringVarP(&groupPrefix, "group-prefix", "", defaultGroupPrefix, "prefix used in group names")
+	generateCmd.Flags().StringVarP(&file, "file", "", "", "file path to save generated topology")
+	generateCmd.Flags().BoolVarP(&deploy, "deploy", "", false, "deploy a fabric based on the generated topology file")
+}
+
+func generateTopologyConfig(name, network, ipv4range, ipv6range string, images map[string]string, licenses map[string]string, nodes ...nodesDef) ([]byte, error) {
+	numStages := len(nodes)
+	config := &clab.Config{
+		Name: name,
+		Topology: clab.Topology{
+			Kinds: make(map[string]clab.NodeConfig),
+			Nodes: make(map[string]clab.NodeConfig),
+		},
+	}
+	config.Mgmt.Network = network
```

Comment:
do we want to render the `mgmt` container if the user did not pass the value?
What if we skip setting the default value for mgmtNetName at L123? That way during the deploy we will apply the necessary defaults

Reply:
the defaults of `--network` in `generate` and `deploy` are the same, so it will be the same result

Reply:
yes, but my point is that it is not really needed to set a network to a default name `clab` here, as this will be done in the backend

i.e. if we change the default name later in deploy, we would need to remember to change the default value of that flag here as well

Reply:
removed the default value